src/test/conftest.py

```diff
@@ -54,3 +74,416 @@ def create_tables():
+    """
+    yield Event()
+
+
+@pytest.fixture
+def comparison_scheduler(request, comparison_finished_event) -> ComparisonScheduler:
+    """Returns an instance of :py:class:`~scheduler.comparison_scheduler.ComparisonScheduler`.
+    The scheduler has some extra testing features. See :py:class:`SchedulerTestConfig` for the features.
+    """
+    test_config = merge_markers(request, 'SchedulerTestConfig', SchedulerTestConfig.get_subclass_from_request(request))
+    _comparison_scheduler = ComparisonScheduler()
+
+    _comparison_scheduler.db_interface = test_config.comparison_db_class()
+
+    def _callback_hook():
+        comparison_finished_event.set()
+
+    _comparison_scheduler.callback = _callback_hook
+
+    if test_config.start_processes:
+        _comparison_scheduler.start()
+
+    yield _comparison_scheduler
+
+    if test_config.start_processes:
+        _comparison_scheduler.shutdown()
+
+
+@dataclass
+class SchedulerTestConfig:
+    """A declarative class that describes the desired behavior for the fixtures :py:func:`~analysis_finished_event`,
+     :py:func:`unpacking_scheduler` and :py:func:`comparison_scheduler`.
+
+    The fixtures don't do any assertions, they MUST be done by the test using the fixtures.
+    """
+
+    #: The amount of items that the :py:class:`~scheduler.analysis.AnalysisScheduler` must analyze before
+    #: :py:func:`analysis_finished_event` gets set.
+    items_to_analyze: int = 0
+    #: Set the class that is used as :py:class:`~storage.db_interface_backend.BackendDbInterface`.
+    #: This can be either a mocked class or the actual :py:class:`~storage.db_interface_backend.BackendDbInterface`.
+    #: This is used by the :py:func:`analysis_scheduler`
+    backend_db_class: Type = BackEndDbInterfaceMock
+    #: Set the class that is used as :py:class:`~storage.db_interface_comparison.ComparisonDbInterface`.
+    #: This can be either a mocked class or the actual :py:class:`~storage.db_interface_comparison.ComparisonDbInterface`.
+    #: This is used by the :py:func:`comparison_scheduler`
+    comparison_db_class: Type = ComparisonDbInterface
+    #: Set the class that is used as :py:class:`~storage.fsorganizer.FSOrganizer`.
+    #: This can be either a mocked class or the actual :py:class:`~storage.fsorganizer.FSOrganizer`.
+    #: This is used by the :py:func:`unpacking_scheduler`
+    fs_organizer_class: Type = FSOrganizerMock
+    #: If set to ``True`` the :py:func:`unpacking_scheduler` and :py:func:`analysis_scheduler` are connected via their
+    #: hooks.
+    #: To be percise the analysis is started after unpacking.
+    #: Also the objects to be analysed and the finished analysis is added to the backend database.
+    pipeline: bool = False
+    #: If set to ``False`` no processes will be started.
+    start_processes: bool = True
+
+    @staticmethod
+    def get_subclass_from_request(request: pytest.FixtureRequest) -> Type['SchedulerTestConfig']:
```

Comment:
I changed this thing to sth that makes more sense but stil is a bit hacky.

I noticed that it makes no sense to have subclasses for each test type since the fixtures don't care about what type of test this is.
So I removed the subclassing and added some rather hacky looking methods to `SchedulerTestConfig` to provide different sets of defaults.
Do you have any better idea how to archive this?